### ISSUE TITLE
Linker command failed with exit code 1 (use -v to see invocation)

### ISSUE THREAD

**Issue by nunoarruda** [phonegap/phonegap-plugin-push#1240]
### Expected Behaviour

Build success
### Actual Behaviour

Build fails
### Reproduce Scenario (including but not limited to)

```
Cordova CLI: 6.3.1
Ionic Framework Version: 2.0.0-beta.11
Ionic CLI Version: 2.0.0
Ionic App Lib Version: 2.0.0
ios-deploy version: 1.9.0 
ios-sim version: 5.0.8 
OS: macOS Sierra
Node Version: v6.6.0
Xcode version: Xcode 8.0 Build version 8A218a
```
#### Steps to Reproduce
1. `ionic build ios`
2. Open "yourAppFolder"/platforms/ios/yourAppName.xcodeproj
3. Build
#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)

iOS 10.0.2
#### Cordova CLI version and cordova platform version

cordova 6.3.1
ios 4.2.1
#### Plugin version

phonegap-plugin-push 1.8.2
#### Logs taken while reproducing problem

```
ld: warning: object file (/Users/nunoarruda/repos/zee-p3-app/platforms/ios/zee-p3-app/Plugins/phonegap-plugin-push/GCM/Libraries/libProtocolBuffers.a(Descriptor.pb.o)) was built for newer iOS version (8.3) than being linked (8.0)

ld: warning: object file (/Users/nunoarruda/repos/zee-p3-app/platforms/ios/zee-p3-app/Plugins/phonegap-plugin-push/GCM/Libraries/libProtocolBuffers.a(ObjectivecDescriptor.pb.o)) was built for newer iOS version (8.3) than being linked (8.0)

ld: warning: object file (/Users/nunoarruda/repos/zee-p3-app/platforms/ios/zee-p3-app/Plugins/phonegap-plugin-push/GCM/Libraries/libProtocolBuffers.a(PBArray.o)) was built for newer iOS version (8.3) than being linked (8.0)

...

ld: 270 duplicate symbols for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by macdonst** [comment]
@nunoarruda this seems like a duplicate of #1025.

Did you try removing and adding the ios platform?

**Comment by nunoarruda** [comment]
@macdonst Yes, I've tried removing and adding the iOS platform, I've tried `cordova clean ios` and also removing the `platforms` and `plugins` folders and installing everything again. The problem persists.

**Comment by carlosrodriguesjunior** [comment]
@nunoarruda 
I had the same probelm. In my case phonegap-plugin-push conflict with the other plugin cordova-plugin-fcm.

**Comment by nunoarruda** [comment]
@carlosrodriguesjunior thanks for the tip, it was helpful. I've tested phonegap-plugin-push in a blank Ionic project and it worked. Tested it again in my project and it failed. It looks like it's conflicting with [cordova-plugin-googleplus](https://github.com/EddyVerbruggen/cordova-plugin-googleplus):

<img width="971" alt="screen shot 2016-10-09 at 15 04 07" src="https://cloud.githubusercontent.com/assets/1189149/19221375/b9820c1a-8e31-11e6-8e3b-584edf293f83.png">

I have no alternative to cordova-plugin-googleplus so I have to keep using it. Will try [OneSignal-Cordova-SDK](https://github.com/OneSignal/OneSignal-Cordova-SDK) instead of phonegap-plugin-push.

**Comment by macdonst** [comment]
@nunoarruda ah, I see. Well once PR #1183 is merged the plugin will be using CocoaPods to pull in those dependencies. Then @EddyVerbruggen and I will be able to resolve the issue.

**Comment by macdonst** [comment]
Waiting for https://github.com/EddyVerbruggen/cordova-plugin-googleplus/issues/272 to be fixed.

**Comment by ramoncarreras** [comment]
Hi, I have the same problem but I don't have _'cordova-plugin-googleplus'_ plugin installed in my APP:


```
cordova-plugin-compat 1.1.0 "Compat"
cordova-plugin-console 1.0.5 "Console"
cordova-plugin-contacts 2.2.1 "Contacts"
cordova-plugin-device 1.1.4 "Device"
cordova-plugin-dialogs 1.3.1 "Notification"
cordova-plugin-facebook4 1.7.4 "Facebook Connect"
cordova-plugin-file 4.3.1 "File"
cordova-plugin-file-transfer 1.6.1 "File Transfer"
cordova-plugin-globalization 1.0.5 "Globalization"
cordova-plugin-inappbrowser 1.6.1 "InAppBrowser"
cordova-plugin-network-information 1.3.1 "Network Information"
cordova-plugin-settings-hook 0.2.4 "Cordova Settings Hooks"
cordova-plugin-splashscreen 4.0.1 "Splashscreen"
cordova-plugin-statusbar 2.2.1 "StatusBar"
cordova-plugin-whitelist 1.3.1 "Whitelist"
cordova-plugin-x-socialsharing 5.1.3 "SocialSharing"
ionic-plugin-deploy 0.6.5 "IonicDeploy"
ionic-plugin-keyboard 2.2.1 "Keyboard"
phonegap-plugin-push 1.9.2 "PushPlugin"
```

How can I know plugin what causing me this build error?

**Comment by adamgalmor** [comment]
same here but with this plugins:

 "com.cmackay.plugins.googleanalytics" 
 "cordova-plugin-console"
 "cordova-plugin-device" 
 "cordova-plugin-inappbrowser" 
 "cordova-plugin-ms-appinsights"
 "cordoba-plugin-device"
 "cordova-plugin-splashscreen" 
 "cordova-plugin-statusbar" 
 "cordova-plugin-whitelist" 
 "ionic-plugin-keyboard" 
 "phonegap-plugin-push" 
`

**Comment by ramoncarreras** [comment]
Hi @adamgalmor ,
the solution for me was open the **.xcworkspace file insted of .xcodeproj** located inside my '/platforms/ios' folder.

**Comment by adamgalmor** [comment]
@ramoncarreras thanks!!!

**Comment by serpentinefire** [comment]
@nunoarruda I have the same problem. Did you ever get a solution or work around?

**Comment by Mahluleli-Goodson** [comment]
@ramoncarreras dude you saved a brother!

**Comment by macdonst** [comment]
@serpentinefire go vote up EddyVerbruggen/cordova-plugin-googleplus#272 so Eddy knows it's important to fix.

@Mahluleli-Goodson XCode Workspaces were introduced in 2011. You are always supposed to use them over Projects when available but thanks to @ramoncarreras for commenting to remind folks.

**Comment by Mahluleli-Goodson** [comment]
@macdonst thanks for that

**Comment by russiann** [comment]
@ramoncarreras thanks!!!

**Comment by nailujpeloz** [comment]
Ey @ramoncarreras thanks for your help!!!

**Comment by mrakowski0** [comment]
Thanks, a life saver! 👍

**Comment by wcustodio** [comment]
Thanks @ramoncarreras!! I can sleep now!

**Comment by YaroslavG** [comment]
I used OneSignal-Cordova-SDK and had the same problem! Thanks @ramoncarreras, you saved my time! :)

**Comment by ibnclaudius** [comment]
@ramoncarreras Thanks! Perfect!

**Comment by speedtreammanga** [comment]
@ramoncarreras Awesome, thank you!

**Comment by ramoncarreras** [comment]
You're welcome @adamgalmor, @Mahluleli-Goodson, @russiann, @nailujpeloz, @wcustodio, @YaroslavG, @ibnclaudius and @speedtreammanga.

**Comment by andrews05** [comment]
I just encountered this problem today and it was only fixed after removing cordova-plugin-console. Weird because everything was working fine last week.

**Comment by fredgalvao** [comment]
@andrews05 Did you create your project recently? If so, can you confirm if you created it using `cordova-ios`@4.5.0? If so, I'd recomend you read the to-be-released blog post on that new version [here](https://github.com/apache/cordova-docs/pull/727), to understand why a duplicate symbol error is related to the `cordova-plugin-console` plugin.

**Comment by fredgalvao** [comment]
@andrews05  I see... Create a new issue with the full template report then, please.

**Comment by mglmgl** [comment]
**Important (since cordova-ios 4.5.0):**

Please note, that the cordova plugin cordova-plugin-console is now part of the cordova-ios platform. So if you're using cordova-ios@4.5.0 or higher absolutely remove cordova-plugin-console!

`cordova plugin rm cordova-plugin-console`

This concerns plain cordova apps and ionic apps as well. 
Further information: [Console-Plugin on GitHub](https://github.com/apache/cordova-plugin-console/blob/master/README.md)

**Comment by AbuzerAsif** [comment]
thanks @mglmgl 

it worked for me 👍

**Comment by nickgrealy** [comment]
For me, I had to remove all plugins, then re-add them (and re-build the project) one-by-one to identify the conflicting library.

**Comment by henriqueweiand** [comment]
thanks @mglmgl

it worked for me 👍
